src/avatar_action.cpp

```diff
@@ -361,7 +416,8 @@ bool avatar_action::move( avatar &you, map &m, const tripoint &d )
                 return false;
             }
         }
-        if( ( fromSwimmable && fromDeepWater && !fromBoat ) || query_yn( _( "Dive into the water?" ) ) ) {
+        if( ( fromSwimmable && fromDeepWater && !fromBoat ) ||
+            ( query_yn( _( "Dive into the water?" ) ) && check_water_affect_items( you ) ) ) {
```

Comment:
If I understand correctly, this will display two confirmation dialogs.  It might be better UX if the water damage information could be folded into the existing `"Dive into the water?"` confirmation, if you can think of a nice way to implement that.